subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/r34/BuildProgressCrossVersionSpec.groovy

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r34
+
+import org.gradle.integtests.tooling.fixture.ProgressEvents
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.tooling.BuildException
+import org.gradle.tooling.ProjectConnection
+
+@ToolingApiVersion(">=2.5")
+@TargetGradleVersion(">=3.4")
+class BuildProgressCrossVersionSpec extends ToolingApiSpecification {
```

Comment:
Something that we're missing is an operation that represents the execution of a build script, so that in the case of a build script that applies another build script that then applied a plugin, you can tell where the plugin came from.

In general, I think we should start by adding build operations for coarse grained work before adding build operations for fine-grained work.